server/rooms.ts

```diff
@@ -473,23 +473,12 @@ export abstract class BasicRoom {
 		if (!this.settings.modjoin) return true;
 		// users with a room rank can always join
 		if (this.auth.has(user.id)) return true;
-		const userGroup = user.can('makeroom') ? user.group : this.auth.get(user.id);
 
 		const modjoinSetting = this.settings.modjoin !== true ? this.settings.modjoin : this.settings.modchat;
 		if (!modjoinSetting) return true;
-		let modjoinGroup = modjoinSetting;
+		if (!(modjoinSetting in Config.groups)) throw new Error(`Invalid modjoin setting in ${this.roomid}: ${modjoinSetting}`);
```

Comment:
This prevents `trusted` and `autoconfirmed` to be used. I might get rid of this `throw` entirely (maybe replace it with `Monitor.error`); it's probably bad form to crash the server because you removed a group from the config file.

Reply:
Maybe with a `isAuthLevel` function in `Users.Auth`?